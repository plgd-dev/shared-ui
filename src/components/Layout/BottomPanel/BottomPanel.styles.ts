import { css } from '@emotion/react'
import { COLLAPSE_ANIMATION_TIME } from '../constants'
import { panelSizes } from '../LeftPanel/constants'
import { ThemeType, getThemeColor, getTheme } from '../../Atomic/_theme'
import { fontPrimary } from '../../Atomic/_utils/commonStyles'

export const selectionPanel = css`
    position: fixed;
    left: ${panelSizes.FULL}px;
    bottom: 0;
    right: 0;
    z-index: 20;
    transition: all ${COLLAPSE_ANIMATION_TIME}ms;
`

export const isLeftPanelCollapsed = css`
    left: ${panelSizes.COLLAPSED}px;
`

export const iframeMode = css`
    //left: 0;
`

export const absolute = css`
    position: absolute;
`

export const inner = (theme: ThemeType) => css`
    padding: 20px 50px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    flex-wrap: wrap;
    background: ${getThemeColor(theme, `BottomPanel.background`)};
    box-shadow: 0 -30px 20px rgba(28, 52, 99, 0.05);
    transition: all 0.35s;
`

export const left = css`
    display: flex;
    align-items: center;
`

export const label = (theme: ThemeType) => css`
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    color: ${getThemeColor(theme, `BottomPanel.value.color`)};
`

export const selectionInfo = (theme: ThemeType) => css`
    display: flex;
    align-items: center;
    font-family: ${getTheme(theme, `Global.fontPrimary`, fontPrimary)};
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 24px;
    color: ${getThemeColor(theme, `BottomPanel.value.color`)};
`

export const right = css`
    margin: 0 -4px;
    display: flex;
    align-items: center;

    & > button {
        margin: 0 4px;
    }
`
