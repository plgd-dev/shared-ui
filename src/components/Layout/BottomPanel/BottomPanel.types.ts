import { ReactNode } from 'react'

export type Props = {
    actionPrimary?: ReactNode
    actionSecondary?: ReactNode
    attribute: string
    iframeMode?: boolean
    leftPanelCollapsed?: boolean
    value: string | number
    show: boolean
}
