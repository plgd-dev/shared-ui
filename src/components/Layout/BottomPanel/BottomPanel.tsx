import { FC } from 'react'
import { motion, AnimatePresence } from 'framer-motion'

import { Props } from './BottomPanel.types'
import * as styles from './BottomPanel.styles'

const BottomPanel: FC<Props> = (props) => {
    const { actionPrimary, actionSecondary, attribute, value, iframeMode, leftPanelCollapsed, show } = props
    return (
        <AnimatePresence>
            {show && (
                <motion.div
                    animate={{ opacity: 1, y: 0 }}
                    css={[styles.selectionPanel, leftPanelCollapsed && styles.isLeftPanelCollapsed, iframeMode && styles.iframeMode]}
                    exit={{
                        opacity: 0,
                        y: 150,
                    }}
                    initial={{ opacity: 0, y: 150 }}
                    transition={{
                        duration: 0.3,
                    }}
                >
                    <div css={styles.inner}>
                        <div css={styles.left}>
                            {!!attribute && <div css={styles.label}>{attribute}:&nbsp;</div>}
                            {!!value && <div css={styles.selectionInfo}>{value}</div>}
                        </div>
                        <div css={styles.right}>
                            {actionSecondary}
                            {actionPrimary}
                        </div>
                    </div>
                </motion.div>
            )}
        </AnimatePresence>
    )
}

BottomPanel.displayName = 'BottomPanel'

export default BottomPanel
