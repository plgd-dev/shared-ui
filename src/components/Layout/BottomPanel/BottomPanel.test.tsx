import { render } from '../../../../test/jest-wrapper'
import BottomPanel from './BottomPanel'

describe('<BottomPanel>', () => {
    it('renders attribute and value when provided', () => {
        const { getByText, asFragment } = render(<BottomPanel show attribute='Test Attribute' value='Test Value' />)
        expect(getByText('Test Attribute:')).toBeInTheDocument()
        expect(getByText('Test Value')).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })

    it('does not render attribute and value when not provided', () => {
        const { queryByText } = render(<BottomPanel show />)
        expect(queryByText('Test Attribute:')).toBeNull()
        expect(queryByText('Test Value')).toBeNull()
    })

    it('renders actionPrimary and actionSecondary when provided', () => {
        const actionPrimary = <button>Primary Action</button>
        const actionSecondary = <button>Secondary Action</button>
        const { getByText } = render(<BottomPanel show actionPrimary={actionPrimary} actionSecondary={actionSecondary} />)
        expect(getByText('Primary Action')).toBeInTheDocument()
        expect(getByText('Secondary Action')).toBeInTheDocument()
    })
})
