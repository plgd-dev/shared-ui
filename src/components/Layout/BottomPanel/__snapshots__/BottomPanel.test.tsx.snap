// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomPanel> renders attribute and value when provided 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: fixed;
  left: 280px;
  bottom: 0;
  right: 0;
  z-index: 20;
  -webkit-transition: all 300ms;
  transition: all 300ms;
}

.emotion-1 {
  padding: 20px 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  background: #FFFFFF;
  box-shadow: 0 -30px 20px rgba(28, 52, 99, 0.05);
  -webkit-transition: all 0.35s;
  transition: all 0.35s;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  color: #0A2965;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 600;
  font-size: 16px;
  line-height: 24px;
  color: #0A2965;
}

.emotion-5 {
  margin: 0 -4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5>button {
  margin: 0 4px;
}

<div
    class="emotion-0"
    style="opacity: 0; transform: translateY(150px) translateZ(0);"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          Test Attribute:Â 
        </div>
        <div
          class="emotion-4"
        >
          Test Value
        </div>
      </div>
      <div
        class="emotion-5"
      />
    </div>
  </div>
</DocumentFragment>
`;
